src/api/Controllers/WaterIngressController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using BDMS.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace BDMS.Controllers;
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class WaterIngressController : BdmsControllerBase<WaterIngress>
+{
+    private readonly BdmsContext context;
+
+    public WaterIngressController(BdmsContext context, ILogger<WaterIngress> logger)
+        : base(context, logger)
+    {
+        this.context = context;
+    }
+
+    /// <summary>
+    /// Asynchronously gets all water ingress records optionally filtered by <paramref name="boreholeId"/>.
+    /// </summary>
+    /// <param name="boreholeId"></param>
+    [HttpGet]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    public async Task<IEnumerable<WaterIngress>> GetAsync([FromQuery] int? boreholeId = null)
+    {
+        var waterIngresses = context.WaterIngresses
+            .AsNoTracking();
```

Comment:
PP: Das hätte ich auf einer Zeile belassen, sieht irgendwie besser aus:
```suggestion
        var waterIngresses = context.WaterIngresses.AsNoTracking();
```